### ISSUE TITLE
Strip trailing whitespace

### ISSUE THREAD

**Issue by HDembinski** [boostorg/serialization#180]
Many headers have trailing white space. This makes writing patches difficult, because my editor automatically strips trailing white space.

Would it be possible to strip the white space once for all headers?

**Comment by robertramey** [comment]
hmm how much white space are we talking about? one or two lines? or ....?  I'm generally not a fan of being overly picky about trivial things, though I consider myself pretty picky.  I don't know how this creates a problem

**Comment by HDembinski** [comment]
I am just talking about so-called trailing white space. The line formatting remains exactly the same, visually there will be no difference, but many lines have extra spaces at the end. These extra  spaces do not contribute to the visual style of the code and can be removed.

I made a PR #181 with all the trailing white-space stripped, but it will change almost every file.

**Comment by HDembinski** [comment]
I personally don't care so much about invisible white space either, but some people care a lot about it, see
https://softwareengineering.stackexchange.com/questions/121555/why-is-trailing-whitespace-a-big-deal

For me personally, it is mostly Linus' fault. He made it so that the diff command in git shows trailing white space as annoying fat red blocks, which look very irritating. Atom, the editor which I use, comes with a built-in functionality to strip trailing white space on save.

In other words, these white space strippers become standard and automatically strip white space from files in Boost.Serialization that I edit. Having to remove all the white space edits from the patch again is extra work.

**Comment by robertramey** [comment]
OK - right now I'm the middle of the first of two very subtle patches.  I was going to tell you to wait.  But now I'm thinking about you'll then have to eliminate the spaces again after I upload.  So the best will be for me to merge the PR right here and then pull the changes.  I'll then re-run my tests.  It just turns out that I'm working on serialization as we speak otherwise it would be a big delay.  Of course I don't have to tell you that if I pull and rerun my tests and something is broken, I'm going to be really, really pissed.  Just consider this.
